docs/core/config/auth/passdb.md

```diff
@@ -442,6 +445,35 @@ a load spike of everybody getting logged in at exactly the same time.
 
 Do not perform any authentication, just store extra fields if user is found.
 
+##### `check_client_fp`
+
+Match client certificate or public key fingerprint. See [[setting,ssl_peer_certificate_fingerprint_hash]]
+
+This is intended to be used instead of verifying certificates with CA certificates. If you are using CA certificates,
+you can only fail authentication if the value is non-empty and does not match. If CA certificates are not used, then
+if no passdb matches the fingerprint, then authentication is failed.
+
+See [[setting,ssl_server_request_client_cert]] on how to allow any certificate.
+
+The failure is per-passdb, so next passdb can continue authenticating the user.
+
+::warning If CA certificates are not used, a passdb must provide a valid check_client_fp (or variant) to validate the
+          client certificate. If none is provided, the authentication will fail.
+
+This also requires [[setting,auth_ssl_require_client_cert,yes]] to work.
```

Comment:
That doesn't seem to be true. As long as a certificate is provided, passdb passes even with auth_ssl_require_client_cert=no. Should we change it in the code that it really is required to prevent misconfigurations?

Reply:
if auth_ssl_require_client_cert, then nothing *enforces* that the cert is valid.